### PR TITLE
Per-class loss weighting (`class_weights`)

### PR DESCRIPTION
**Pull request by Kag-Bar**
Adds an optional `class_weights` parameter to YOLO training that amplifies box/DFL loss for specified classes, reducing False Negatives on important/rare classes.

## Why
In datasets with imbalanced class importance — where some classes are rare but must not be missed — standard equal-weight training doesn't provide enough gradient signal for those classes. This gives users direct control via a simple config parameter.

## How it works
- User provides `class_weights={"cone": 5.0, "person": 1.0}` (dict or list). Unspecified classes default to 1.0.
- The trainer resolves class names → index-ordered weight tensor and logs the mapping.
- In `v8DetectionLoss.get_assigned_targets_and_loss`, `target_scores` are multiplied by per-class weights **before** being used as box/DFL loss weights. BCE classification loss is intentionally unchanged.
- `Metric.mean_results()` uses `np.average(..., weights=class_weights)` for P/R/mAP when weights are set, so `fitness()` favors the best model for important classes.
- When `class_weights` is not set, all code paths are identical to before (zero overhead).

## Files changed
1. `ultralytics/cfg/default.yaml` | Added `class_weights:` parameter 
2. `ultralytics/cfg/__init__.py` | Added `class_weights_resolved` to allowed custom keys
3. `ultralytics/models/yolo/detect/train.py` | Added `_resolve_class_weights()` — resolves dict/list to index-ordered weights, logs them
4. `ultralytics/utils/loss.py` | `v8DetectionLoss.__init__`: stores `(nc,)` weight tensor; `get_assigned_targets_and_loss`: applies weights to box/DFL loss
5.  `ultralytics/utils/metrics.py` | `Metric`: added `class_weights` param + weighted `mean_results()`; propagated through `DetMetrics`, `PoseMetrics`, `SegmentMetrics`, `OBBMetrics`
6. `tests/test_class_weights.py` | 18 unit tests — metrics, loss tensor, trainer resolution, edge cases 
7. `scripts/test_class_weights.py` | Demo: trains baseline vs weighted on coco8, compares per-class recall 

## Usage
```python
model = YOLO("yolo11n.pt")
model.train(
    data="my_data.yaml",
    epochs=100,
    class_weights={"cone": 5.0, "person": 1.0}
    # classes not listed default to 1.0
)
```

## Test results
All 18 unit tests pass. Demo script on coco8 with `class_weights={"person": 10.0}` showed person recall improving from 0.60 → 0.70.

## Release Notes

**New Features**
- Added `class_weights` configuration option for per-class loss weighting on box/DFL losses and fitness calculations; accepts dict or list format with unlisted classes defaulting to 1.0.
- Added `mask_ratio` configuration option for segmentation mask downsampling control.

### CODE REVIEW

**Review by LeDuySon — APPROVED** [review]
Look good to me
